kernel/posix/pthread_common.c

```diff
@@ -18,3 +19,34 @@ void ready_one_thread(_wait_q_t *wq)
 		_ready_thread(th);
 	}
 }
+
+s64_t timespec_to_timeoutms(const struct timespec *abstime)
+{
+	s64_t milli_secs;
+	s32_t secs;
+	struct timespec curtime;
+
+	/* FIXME: Zephyr does have CLOCK_REALTIME to get time.
+	 * As per POSIX standard time should be calculated wrt CLOCK_REALTIME.
+	 * Zephyr deviates from POSIX 1003.1 standard on this aspect.
+	 */
+	clock_gettime(CLOCK_MONOTONIC, &curtime);
+	secs = abstime->tv_sec - curtime.tv_sec;
+
+	if (abstime->tv_sec < curtime.tv_sec ||
```

Comment:
Simplify expression to "if(secs <= 0 || abstime->tv_nsec <= curtime.tv_nsec)"

Reply:
I think it should be "if(secs < 0 || ((secs == 0) && (abstime->tv_nsec <= curtime.tv_nsec))).

Just about condition , :- 
secs = 0 and curtime.tv_nsec = 000000 and abstime->tv_nsec = 999999999
It is a valid condition and it should set timeout.